pkg/console/controllers/route/controller.go

```diff
@@ -57,10 +64,13 @@ type RouteSyncController struct {
 }
 
 func NewRouteSyncController(
+	// top level config
+	configClient configclientv1.ConfigV1Interface,
 	// clients
 	operatorConfigClient operatorclientv1.ConsoleInterface,
 	routev1Client routeclientv1.RoutesGetter,
 	configMapClient coreclientv1.ConfigMapsGetter,
+	secretClient coreclientv1.SecretsGetter,
```

Comment:
Nit: We probably could fix this so that `NewRouteSyncController()` accepts the `KubeV1Client` once, but then separates them out when we create the controller:

```golang
ctrl := &RouteSyncController{
   configMapClient:   KubeV1Client.ConfigMaps()
   secretsClient:  KubeV1Client.Secrets()
```
But I see our other controllers have the same issue.  It would be a tad cleaner if:

```golang
// We are a little incorrect as the secrets client is 1 step into the KubeV1Client as .Secrets
c.secretClient.Secrets(api.OpenShiftConfigNamespace).Get(c.ctx, operatorConfig.Spec.Route.Secret.Name, metav1.GetOptions{})
// This would be ideal
c.secretClient(api.OpenShiftConfigNamespace).Get(c.ctx, operatorConfig.Spec.Route.Secret.Name, metav1.GetOptions{})
```

Reply:
So we can chip away at it, or just fix them all as a separate PR.

Reply:
would rather deal with it in one PR, if thats ok :)